src/services/refactors/generateGetAccessorAndSetAccessor.ts

```diff
@@ -0,0 +1,233 @@
+        accessorName: string;
+    }
+
+    function getAvailableActions(context: RefactorContext): ApplicableRefactorInfo[] | undefined {
+        const { file, startPosition } = context;
+        if (!getConvertibleFieldAtPosition(file, startPosition)) return undefined;
+
+        return [{
+            name: actionName,
+            description: actionDescription,
+            actions: [
+                {
+                    name: actionName,
+                    description: actionDescription
+                }
+            ]
+        }];
+    }
+
+    function getEditsForAction(context: RefactorContext, _actionName: string): RefactorEditInfo | undefined {
+        const { file, startPosition } = context;
+
+        const fieldInfo = getConvertibleFieldAtPosition(file, startPosition);
+        if (!fieldInfo) return undefined;
+
+        const changeTracker = textChanges.ChangeTracker.fromContext(context);
+        const { isStatic, fieldName, accessorName, classLikeContainer, declaration } = fieldInfo;
+
+        const accessorModifiers = getAccessorModifiers(declaration, isStatic);
+        const fieldModifiers = getFieldModifiers(isStatic);
+
+        const getAccessor = generateGetAccessor(fieldName, accessorName, accessorModifiers, declaration, isStatic, classLikeContainer);
+        const setAccessor = generateSetAccessor(fieldName, accessorName, accessorModifiers, declaration, isStatic, classLikeContainer);
+
+        updateFieldDeclaration(changeTracker, file, declaration, fieldName, fieldModifiers, classLikeContainer);
+
+        insertAccessor(changeTracker, file, getAccessor, declaration, classLikeContainer);
+        insertAccessor(changeTracker, file, setAccessor, declaration, classLikeContainer);
+
+        const edits = changeTracker.getChanges();
+        const renameFilename = file.fileName;
+        const renameLocation = getRenameLocation(edits, renameFilename, fieldName, /*isDeclaredBeforeUse*/ false);
+        return { renameFilename, renameLocation, edits };
+    }
+
+    function getUniqueName(baseName: string, fileText: string): string {
+        let nameText = baseName;
+        for (let i = 1; stringContains(fileText, nameText); i++) {
+            nameText = `${baseName}_${i}`;
+        }
+        return nameText;
+    }
+
+    function getRenameLocation(edits: ReadonlyArray<FileTextChanges>, renameFilename: string, fieldName: string, isDeclaredBeforeUse: boolean): number {
+        let delta = 0;
+        let lastPos = -1;
+        for (const { fileName, textChanges } of edits) {
+            Debug.assert(fileName === renameFilename);
+            for (const change of textChanges) {
+                const { span, newText } = change;
+                const index = newText.indexOf(fieldName);
+                if (index !== -1) {
+                    lastPos = span.start + delta + index;
+
+                    // If the reference comes first, return immediately.
+                    if (!isDeclaredBeforeUse) {

```

Comment:
do not think you need this variable.. it is always `false`.

Reply:
can you move this to a utility function and use it in both refactorings instead?